src/login.rs

```diff
@@ -147,6 +149,21 @@ impl Login {
         let self_ = imp::Login::from_instance(self);
         self_.client_id.set(client_id);
 
+        // Before transitioning to any login state or to the chat view, set the empty page as the
+        // main stack's visible child.
+        // This will prevent the login leaflet from bouncing back if the following sequence
```

Comment:
I think this comment block don't suggest enough why we're showing an empty page in the first place. I'd suggest replacing it with something like: 

```rust
// We don't know what login page to show at this point, so we show an empty page until we receive an AuthenticationState that will eventually show the related login page.
```

Reply:
Indeed, that sounds much simpler. 
I also removed the rest of the comment because I think your statement is explains it enough.